server/src/main/java/org/opensearch/identity/Principal.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.common.io.stream.NamedWriteable;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * A principal identifier (user or a group) for extensions requests
+ * The implementation should be serializable/deserializable
+*/
+
+// TODO: Determine how this principal can be shared between threads and
+//  the effects of extending NamedWriteable class
```

Comment:
![image](https://user-images.githubusercontent.com/2754967/187307608-b3e763ab-f30c-44f9-8126-7a1a3ac70c3c.png)

Reply:
I'm not certain the implications of using extending NamedWritable - is there a good reference or implementation that I could follow?

Reply:
I'm not sure about that yet.. but you can search for NamedWritable in the codebase. It is for arbitrary seralizable objects. See interface definition [here](https://github.com/dbwiddis/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/io/stream/NamedWriteable.java)